commit faf29d7f91477c25c85d9d7165d90e8d8f1c512e
Author: Bram Moolenaar <Bram@vim.org>
Date:   Sun Jul 9 11:07:16 2017 +0200

    patch 8.0.0703: illegal memory access with empty :doau command
    
    Problem:    Illegal memory access with empty :doau command.
    Solution:   Check the event for being out of range. (James McCoy)

Index: vim-7.4.488/src/fileio.c
===================================================================
--- vim-7.4.488.orig/src/fileio.c
+++ vim-7.4.488/src/fileio.c
@@ -8716,7 +8716,7 @@ do_doautocmd(arg, do_msg)
     /*
      * Loop over the events.
      */
-    while (*arg && !vim_iswhite(*arg))
+    while (*arg && !ends_excmd(*arg) && !vim_iswhite(*arg))
 	if (apply_autocmds_group(event_name2nr(arg, &arg),
 				      fname, NULL, TRUE, group, curbuf, NULL))
 	    nothing_done = FALSE;
@@ -9224,7 +9224,8 @@ apply_autocmds_group(event, fname, fname
      * Quickly return if there are no autocommands for this event or
      * autocommands are blocked.
      */
-    if (first_autopat[(int)event] == NULL || autocmd_blocked > 0)
+    if (event == NUM_EVENTS || first_autopat[(int)event] == NULL
+	    || autocmd_blocked > 0)
 	goto BYPASS_AU;
 
     /*
@@ -9297,7 +9298,7 @@ apply_autocmds_group(event, fname, fname
     {
 	if (event == EVENT_COLORSCHEME)
 	    autocmd_fname = NULL;
-	else if (fname != NULL && *fname != NUL)
+	else if (fname != NULL && !ends_excmd(*fname))
 	    autocmd_fname = fname;
 	else if (buf != NULL)
 	    autocmd_fname = buf->b_ffname;
Index: vim-7.4.488/src/testdir/test13.in
===================================================================
--- vim-7.4.488.orig/src/testdir/test13.in
+++ vim-7.4.488/src/testdir/test13.in
@@ -48,6 +48,7 @@ otestje3
 :au BufWipeout Xtestje1 buf Xtestje1
 :bwipe
 :w >>test.out
+:doau \|
 :qa!
 ENDTEST
 
Index: vim-7.4.488/src/version.c
===================================================================
--- vim-7.4.488.orig/src/version.c
+++ vim-7.4.488/src/version.c
@@ -1733,6 +1733,8 @@ static int included_patches[] =
 static char *(extra_patches[]) =
 {   /* Add your patch description below this line */
 /**/
+    "8.0.0703",
+/**/
     "8.0.0378",
 /**/
     "8.0.0377",
